Zend/zend_compile.c

```diff
@@ -6893,10 +6893,12 @@ static zend_bool zend_try_ct_eval_array(zval *result, zend_ast *ast) /* {{{ */
 		}
 
 		zend_eval_const_expr(&elem_ast->child[0]);
-		zend_eval_const_expr(&elem_ast->child[1]);
+		if (elem_ast->kind != ZEND_AST_UNPACK) {
```

Comment:
I'd suggest to move all the code dealing with ZEND_AST_UNPACK into a single conditional here at the start, to avoid the need to interleave it in multiple places.